driver-sync/src/test/functional/com/mongodb/client/AtlasDataLakeTest.java

```diff
@@ -0,0 +1,337 @@
+    private void executeOperations(final BsonArray operations) {
+        for (BsonValue cur : operations) {
+            final BsonDocument operation = cur.asDocument();
+            String operationName = operation.getString("name").getValue();
+            BsonValue expectedResult = operation.get("result");
+            String receiver = operation.getString("object").getValue();
+
+            try {
+                BsonDocument actualOutcome = helper.getOperationResults(operation, null);
+                if (expectedResult != null) {
+                    BsonValue actualResult = actualOutcome.get("result");
+                    if (actualResult.isDocument()) {
+                        if (((BsonDocument) actualResult).containsKey("recoveryToken")) {
+                            ((BsonDocument) actualResult).remove("recoveryToken");
+                        }
+                    }
+
+                    assertEquals("Expected operation result differs from actual", expectedResult, actualResult);
+                }
+                assertFalse(String.format("Expected error '%s' but none thrown for operation %s",
+                        getErrorContainsField(expectedResult), operationName), hasErrorContainsField(expectedResult));
+                assertFalse(String.format("Expected error code '%s' but none thrown for operation %s",
+                        getErrorCodeNameField(expectedResult), operationName), hasErrorCodeNameField(expectedResult));
+            } catch (RuntimeException e) {
+                if (!assertExceptionState(e, expectedResult, operationName)) {
+                    throw e;
+                }
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    @Parameterized.Parameters(name = "{0}: {1}")
+    public static Collection<Object[]> data() throws URISyntaxException, IOException {
+        List<Object[]> data = new ArrayList<Object[]>();
+        for (File file : JsonPoweredTestHelper.getTestFiles("/atlas-data-lake")) {
+            BsonDocument testDocument = JsonPoweredTestHelper.getTestDocument(file);
+            for (BsonValue test : testDocument.getArray("tests")) {
+                data.add(new Object[]{file.getName(), test.asDocument().getString("description").getValue(),
+                        testDocument.getString("database_name", new BsonString(getDefaultDatabaseName())).getValue(),
+                        testDocument.getString("collection_name", new BsonString("test")).getValue(),
+                        testDocument.getArray("data", new BsonArray()), test.asDocument(), skipTest(testDocument, test.asDocument())});
+            }
+        }
+        return data;
+    }
+
+    protected MongoClient createMongoClient(final MongoClientSettings settings) {
+        return MongoClients.create(settings);
+    }
+
+    private boolean assertExceptionState(final RuntimeException e, final BsonValue expectedResult, final String operationName) {
```

Comment:
There's a lot of code here.  Was it copied from somewhere else?  Perhaps it can be shared.

Reply:
I have updated this to extend `AbstractUnifiedTest`, but I need the ADL connection string. So I have overloaded `setUp` and made some of the class variables `protected` in `AbstractUnifiedTest`.